### PR TITLE
Show a way to provide a default role

### PR DESCRIPTION
**Pull request by timneedham**
I was being a bit slow and hit this issue just now (having `authErrors` after `postgraphile`... which you might end-up doing if you're following things step-by-step?). Dunno, might help?

### CODE REVIEW

File: src/pages/postgraphile/jwk-verification.md [hunk 1]
```diff
@@ -201,6 +201,29 @@ appropriate and succinct response:
...
+
+```javascript
+const authErrors = (err, req, res, next) => {
+  if (err.name === "UnauthorizedError") {
+    console.log(err); // You will still want to log the error...
+    if (err.code === 'credentials_required') {
+      // No token provided, so assign a default PostgreSQL role (see pgDefaultRole) and continue
+      req.user = {
+        role: 'unauthenticated'
+      }
```
Comment (benjie) [inline src/pages/postgraphile/jwk-verification.md]: I would set `req.user = null` here (there is no user) and interpret the null result in pgSettings

File: src/pages/postgraphile/jwk-verification.md [hunk 2]
```diff
@@ -201,6 +201,29 @@ appropriate and succinct response:
...
+      next()
+    } else {
+      // Some other auth problem, send back as normal
+      res.status(err.status).json({ errors: [{ message: err.message }] });
```
Comment (benjie) [inline src/pages/postgraphile/jwk-verification.md]: Should we check err.status is a valid status code? What should it fall back to - 403?

File: src/pages/postgraphile/jwk-verification.md (hunk 1 above)
Comment (benjie) [reply src/pages/postgraphile/jwk-verification.md]: Perhaps the express-jwt option `credentialsRequired: false` would achieve this in less code?

```js
app.use(jwt({
  secret: 'hello world !',
  credentialsRequired: false
}));
```

Comment (timneedham) [reply src/pages/postgraphile/jwk-verification.md]: Doh! Missed `credentialsRequired`.... yeah, that's much better!

File: src/pages/postgraphile/jwk-verification.md (hunk 2 above)
Comment (timneedham) [reply src/pages/postgraphile/jwk-verification.md]: Not sure of the internals of `express-jwt` (if you always get a valid `status`)... but falling-back to `403` sounds a very good idea!

Comment (timneedham) [reply src/pages/postgraphile/jwk-verification.md]: The combination of `credentialsRequired: false` and:

```
pgSettings: req => {
  const settings = {}
  if (req.user && req.user.role) {
    settings.role = req.user.role
  } else {
    settings.role = 'unauthenticated'
  }
  return settings
}
```

...works great for me, thanks for your help @benjie !

Comment (timneedham) [reply src/pages/postgraphile/jwk-verification.md]: (and because of that, like you say, there's no need to set `req.user` manually, which must be better!)

**Comment by benjie** [comment]
@timneedham It would still be great to add the learnings above to the docs - do you think you could update this PR to detail this for future readers?

**Comment by timneedham** [comment]
Sorry yes, of course... I'll do that this week!

**Comment by benjie** [comment]
Hey @timneedham; just a gentle nudge :smile:

**Comment by timneedham** [comment]
Hi sorry... been away, hopefully have time this week!

**Comment by benjie** [comment]
I'm in no rush, but since it slipped off the "this week" radar I thought I might bump it again ;)

**Comment by benjie** [comment]
Just a gentle nudge @timneedham :wink:

**Comment by timneedham** [comment]
TBH, I've clean lost this thread now... but we're coming back to auth very soon, so I'll pict it up then, if that's OK?

**Comment by benjie** [comment]
Sure 😀

**Comment by benjie** [comment]
@timneedham I'm going to close this for now; but if you intend to address it please feel free to re-open :+1:

**Pull request closed without merging**